torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -1127,6 +1127,18 @@ at::Tensor XLANativeFunctions::dot(const at::Tensor& self,
       << "dot: Expected 1-D argument self, but got " << self.dim() << "-D";
   XLA_CHECK_EQ(tensor.dim(), 1)
       << "dot: Expected 1-D argument tensor, but got " << tensor.dim() << "-D";
+  // Fallback to CPU if both tensor types are integral and atleast one of them
+  // is a long, as this multiplication is not supported for TPUs.
+  XlaDeviceType hw_type =
+      static_cast<XlaDeviceType>(bridge::GetCurrentDevice().type());
+  if (CheckTpuDevice(hw_type) &&
+      (at::isIntegralType(self.scalar_type(), /*include_bool=*/true) &&
+       at::isIntegralType(tensor.scalar_type(), /*include_bool=*/true) &&
+       (at::elementSize(self.scalar_type()) == 8 ||
+        at::elementSize(tensor.scalar_type()) == 8))) {
```

Comment:
this `at::elementSize` is new to me, why couldn't we just check if they are int64? Are you trying to cover the unsigned_int64 case?

Reply:
Yes, also trying to cover the unsigned_int64 case.

Reply:
Can you add a comment here to explain that int64 and uinte64 dot is not supported on TPU yet?